pubsub/subscriptions.js

```diff
@@ -1,187 +1,315 @@
+    callback();
   });
 }
+// [END pubsub_create_push_subscription]
 
-function listAllSubscriptions (callback) {
-  var pubsub = PubSub();
+// [START pubsub_delete_subscription]
+function deleteSubscription (subscriptionName, callback) {
+  // References an existing subscription, e.g. "my-subscription"
+  const subscription = pubsubClient.subscription(subscriptionName);
 
-  // List all subscriptions
-  // See https://googlecloudplatform.github.io/google-cloud-node/#/docs/pubsub/latest/pubsub?method=getSubscriptions
-  pubsub.getSubscriptions(function (err, subscriptions) {
+  // Deletes the subscription
+  subscription.delete((err) => {
     if (err) {
-      return callback(err);
+      callback(err);
+      return;
     }
 
-    console.log('Found %d subscription(s)!', subscriptions.length);
-    return callback(null, subscriptions);
+    console.log(`Subscription ${subscription.name} deleted.`);
+    callback();
   });
 }
+// [END pubsub_delete_subscription]
+
+// [START pubsub_get_subscription_metadata]
+function getSubscriptionMetadata (subscriptionName, callback) {
+  // References an existing subscription, e.g. "my-subscription"
+  const subscription = pubsubClient.subscription(subscriptionName);
+
+  // Gets the metadata for the subscription
+  subscription.getMetadata((err, metadata) => {
+    if (err) {
+      callback(err);
+      return;
+    }
 
-function handleMessage (message) {
-  console.log('received message: ' + message.data);
+    console.log(`Subscription: ${metadata.name}`);
+    console.log(`Topic: ${metadata.topic}`);
+    console.log(`Push config: %s`, metadata.pushConfig.pushEndpoint);
+    console.log(`Ack deadline: ${metadata.ackDeadlineSeconds}s`);
+    callback();
+  });
 }
+// [END pubsub_get_subscription_metadata]
 
+// [START pubsub_pull_messages]
 function pullMessages (subscriptionName, callback) {
-  var pubsub = PubSub();
-  var subscription = pubsub.subscription(subscriptionName);
+  // References an existing subscription, e.g. "my-subscription"
+  const subscription = pubsubClient.subscription(subscriptionName);
 
-  // Pull any messages on the subscription
-  // See https://googlecloudplatform.github.io/google-cloud-node/#/docs/pubsub/latest/pubsub/subscription?method=pull
-  subscription.pull(function (err, messages) {
+  // Pulls messages. Set returnImmediately to false to block until messages are
+  // received.
+  subscription.pull({ returnImmediately: true }, (err, messages) => {
     if (err) {
-      return callback(err);
+      callback(err);
+      return;
     }
-    // Do something for each message
-    messages.forEach(handleMessage);
 
-    console.log('Pulled %d message(s)!', messages.length);
+    console.log(`Received ${messages.length} messages.`);
 
-    var ackIds = messages.map(function (message) {
-      return message.ackId;
+    messages.forEach((message) => {
+      console.log(`* ${message.id} %j %j`, message.data, message.attributes);
```

Comment:
Hmm, I'll pull `message.id` out, because pulling the other two in doesn't work so well.